java/src/jmri/jmrit/beantable/beanedit/BeanItemPanel.java

```diff
@@ -44,20 +45,24 @@ public void setResetItem(AbstractAction reset) {
         this.reset = reset;
     }
 
-    AbstractAction save;
-    AbstractAction reset;
+    private AbstractAction save;
+    private AbstractAction reset;
 
-    ArrayList<BeanEditItem> items = new ArrayList<BeanEditItem>();
+    private final ArrayList<BeanEditItem> items = new ArrayList<>();
 
     public void addItem(BeanEditItem bei) {
         items.add(bei);
     }
 
+    /**
+     * Get Copy of List of Bean Edit Items
+     * @return copy of Array List.
+     */
     public List<BeanEditItem> getListOfItems() {
-        return items;
+        return new ArrayList<>(items);
```

Comment:
It's often better to use [Collections.unmodifiableList(items)](https://docs.oracle.com/javase/1.5.0/docs/api/java/util/Collections.html#unmodifiableList(java.util.List)). That way, you don't need a copy of the list. `unmodifiableList(items)` returns the list, but ensures that the caller cannot change it and therefore it's safe to return the list.

So the code in this case would be:
`return Collections.unmodifiableList(items);`

Reply:
Thanks, that makes sense, will edit this PR.